src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala

```diff
@@ -46,36 +46,59 @@ final case class ThermalGrid(
     * time
     * @param tick
     *   Questioned instance in time
-    * @param ambientTemperature
-    *   Ambient temperature in the instance in question
+    * @param lastAmbientTemperature
+    *   Ambient temperature in the instance in question until actual tick
+    * @param actualAmbientTemperature
+    *   Ambient temperature in the instance in question at tick
     * @param state
     *   Currently applicable state of the thermal grid
     * @return
     *   The total energy demand of the grid
     */
   def energyDemand(
       tick: Long,
-      ambientTemperature: Temperature,
+      // FIXME this is also in state -> refactoring by HiWi
+      lastAmbientTemperature: Temperature,
+      actualAmbientTemperature: Temperature,
```

Comment:
Note from my side: In my opinion both temperatures are necessary here, the ambientTemperature of the last state and the actual one. Since the HpState contains the lastAmbientTemperature this can be done nicer and would be a good HiWi-Job. As well some tests for different cases like falling and rising temperature would be great.

Reply:
Transfered this into #916